component/encryptor/encryptor.go

```diff
@@ -0,0 +1,368 @@
+package encryptor
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/binary"
+	"fmt"
+	"os"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/Azure/azure-storage-fuse/v2/common/config"
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
+	"github.com/Azure/azure-storage-fuse/v2/internal"
+	"github.com/Azure/azure-storage-fuse/v2/internal/handlemap"
+)
+
+// Implements two mount approach with only one file for data and metadata.
+type Encryptor struct {
+	internal.BaseComponent
+	handle           *os.File
+	blockSize        uint64
+	mountPointCipher string
+	encryptionKey    []byte
```

Comment:
Can have an instance of encryptorOptions here instead of reassigning.

Reply:
this was done to avoid reconverting string to bytes every time it is used.